package.json

```diff
@@ -11,14 +11,17 @@
 		"url": "https://sindresorhus.com"
 	},
 	"engines": {
-		"node": ">=8"
+		"node": ">=8.3.0"
 	},
 	"scripts": {
-		"test": "xo && ava && tsd"
+		"test": "xo && tsc --noEmit && nyc ava",
+		"prepare": "npm run build",
+		"build": "del dist && tsc",
+		"prepublishOnly": "npm run build"
```

Comment:
As far as I know `prepare` is meant to replace `prepublishOnly` so you don't need both.

However I've been using the more specific `prepack` which feels like it's the exact moment when files should be built: right before the tar.gz is created.

Reply:
> As far as I know prepare is meant to replace prepublishOnly so you don't need both.

Yes, but it's a mess since old npm version use `prepare`, new ones use `prepublishOnly`, and future ones will use `prepare` but with it will work differently. I'm not sure why we use both here. Maybe look at the git commits/blame.

> However I've been using the more specific prepack which feels like it's the exact moment when files should be built: right before the tar.gz is created.

Does that one also work when users install a Git commit hash directly? `npm i sindresorhus/conf#ad32d2`

Reply:
> prepack: run BEFORE a tarball is packed (on npm pack, npm publish, and when installing git dependencies)

Reply:
Compared with:

> prepare: Run both BEFORE the package is packed and published, on local npm install without any arguments, and when installing git dependencies (See below). This is run AFTER prepublish, but BEFORE prepublishOnly.

Reply:
https://github.com/sindresorhus/conf/commit/be694026e6fa09a3482a4d6fd3f260f07d1e36f6